src/main/java/org/checkerframework/languageserver/CFTextDocumentService.java

```diff
@@ -42,11 +66,15 @@ void setExecutor(CheckExecutor executor) {
      *     href="https://microsoft.github.io/language-server-protocol/specification#textDocument_publishDiagnostics">specification</a>
      */
     private void clearDiagnostics(List<File> files) {
-        files.forEach(
-                file ->
-                        server.publishDiagnostics(
-                                new PublishDiagnosticsParams(
-                                        file.toURI().toString(), Collections.emptyList())));
+        for (File file : files) {
+            RangeMap<ComparablePosition, List<String>> currentTypeInfo = filesToTypeInfo.get(file);
+            if (currentTypeInfo != null) {
+                currentTypeInfo.clear();
```

Comment:
Instead of just clearing the contained map, should (additionally?) the key be removed from `filesToTypeInfo`?
Otherwise, when a lot of files are opened and closed, `filesToTypeInfo` will continue to accumulate keys that map to an empty RangeMap.
Or maybe this should only be done in `didClose`?